src/coreclr/jit/lower.cpp

```diff
@@ -1464,11 +1464,13 @@ GenTree* Lowering::LowerFloatArg(GenTree** pArg, CallArg* callArg)
 //
 GenTree* Lowering::LowerFloatArgReg(GenTree* arg, regNumber regNum)
 {
+    assert(varTypeUsesFloatReg(arg));
+
     var_types floatType = arg->TypeGet();
-    assert(varTypeIsFloating(floatType));
-    var_types intType = (floatType == TYP_DOUBLE) ? TYP_LONG : TYP_INT;
-    GenTree*  intArg  = comp->gtNewBitCastNode(intType, arg);
+    var_types intType   = (floatType == TYP_FLOAT) ? TYP_INT : TYP_LONG;
```

Comment:
Should there be some assert that the `size` of `floatType` isn't greater than 8?

Reply:
It should be part of `gtNewBitCastNode`, though isn't right now. I'll add it in some future change.

Currently it is implied by the `assert(varTypeUsesFloatReg(arg));` above.